### PR TITLE
Correct bfmi denominator (fixes #858)

### LINKED ISSUE #858: Incorrect denominator in bfmi
The original equation for `bfmi` is at the bottom of page 7 in [this paper](https://arxiv.org/pdf/1604.00695v1.pdf). The numerator in numpy for a vector `E` is `(length(E) - 1) * np.mean(np.square(np.diff(E)))`. The denominator is `(length(E) - 1) * np.var(E, ddof = 1)`. The two scaling factors cancel and we get
```python
def bfmi(E):
    return np.mean(np.square(np.diff(E))) / np.var(E, ddof = 1)
```
The current implementation uses `ddof=0`, so it's off from the equation in the paper by `length(E) / (length(E) - 1)`

**Additional context**
PyStan's implementation [here](https://github.com/stan-dev/pystan/blob/cbc5c0fc8597504776c08a138adc9ae63329e6a1/pystan/diagnostics.py#L213-L217) is equivalent to the paper's. RStan's implementation of `get_bfmi` [here](https://rdrr.io/cran/rstan/src/R/check_hmc_diagnostics.R) has a different implementation than either ArviZ's current one or PyStan's. Theirs amounts to the original paper's equation multiplied by `(length(E) - 1) / length(E)`.

### PR DESCRIPTION
**Pull request by nitishp25**
Correction for bfmi formula implementation according to page 7 of [this paper](https://arxiv.org/pdf/1604.00695v1.pdf).

Fixes #858.

- [x] Fix denominator in if statement

- [x] Fix denominator in else statement

- [x] Fix bfmi docs to render example properly

### CODE REVIEW

**Comment by nitishp25** [comment]
can someone tell me why this fails?

**Comment by percygautam** [comment]
Hi @nitishp25 , I think you have to change the test_bfmi() function of test_diagnostics.py. Run the code in test function with updated ddof value and replace the result with the previous value.

**Comment by nitishp25** [comment]
Got it, thanks @percygautam!

**Comment by nitishp25** [comment]
Ready to merge.

File: arviz/stats/diagnostics.py [hunk 1]
```diff
@@ -526,7 +526,7 @@ def _bfmi(energy):
     energy_mat = np.atleast_2d(energy)
     num = np.square(np.diff(energy_mat, axis=1)).mean(axis=1)  # pylint: disable=no-member
     if energy_mat.ndim == 2:
-        den = _numba_var(svar, np.var, energy_mat, axis=1, ddof=0)
+        den = _numba_var(svar, np.var, energy_mat, axis=1, ddof=1)
     else:
         den = np.var(energy, axis=1)
```
Comment (OriolAbril) [inline arviz/stats/diagnostics.py]: I think that numpy's default is `ddof=0`, so this line in the else should be corrected too. Either way, even if the default were `ddof=1` it's probably better to make it explicit in this case.

**Comment by OriolAbril** [comment]
Also, this is a little unrelated but it would be great if you could fix it. I have just seen that [bfmi docs](https://arviz-devs.github.io/arviz/generated/arviz.bfmi.html#arviz.bfmi) are not rendered properly (see [rhat docs](https://arviz-devs.github.io/arviz/generated/arviz.rhat.html#arviz.rhat) to see how they should look like) there is no `Out` section even though there should be and the `In []` is not properly highlighted. A white line between [lines 48 and 49](https://github.com/arviz-devs/arviz/blob/c3158613f9e793902eddd0e5ecd715595c31c824/arviz/stats/diagnostics.py#L48) should fix this.

File: arviz/stats/diagnostics.py (hunk 1 above)
Comment (nitishp25) [reply arviz/stats/diagnostics.py]: Yup, the default is 1. I'll change this. Thanks for the feedback @OriolAbril!

**Comment by nitishp25** [comment]
> Also, this is a little unrelated but it would be great if you could fix it. I have just seen that [bfmi docs](https://arviz-devs.github.io/arviz/generated/arviz.bfmi.html#arviz.bfmi) are not rendered properly (see [rhat docs](https://arviz-devs.github.io/arviz/generated/arviz.rhat.html#arviz.rhat) to see how they should look like) there is no `Out` section even though there should be and the `In []` is not properly highlighted. A white line between [lines 48 and 49](https://github.com/arviz-devs/arviz/blob/c3158613f9e793902eddd0e5ecd715595c31c824/arviz/stats/diagnostics.py#L48) should fix this.

Got it. Working on it.

**Comment by sethaxen** [comment]
This looks like the right fix. With the fix to the else statement suggested by @OriolAbril, this looks good to me. Thanks for the PR!

**Comment by nitishp25** [comment]
Please review @OriolAbril

**Comment by OriolAbril** [comment]
Thanks for you PR!

**Pull request merged**